### ISSUE TITLE
Compiling Issues

### ISSUE THREAD

**Issue by 2020chris** [xphoniex/alphazero-quoridor#1]
I am having some issues compiling and running main.py, as it is saying it can't find the pathFinder module. Can main.py be compiled in both python2 and python3?

**Comment by xphoniex** [comment]
Use python2.7. 

I'm getting an error re `print` when using python3 so I can't reproduce what you're seeing.

Also, in case you get it working, make sure you set [`load_model`](https://github.com/xphoniex/alphazero-quoridor/blob/master/main.py#L17) to `False` on first run.

**Comment by 2020chris** [comment]
Thanks for the reply. I assume I need to re-install torch with python2.7 also? 

Should it be able to start training off the bat by running python main.py? Or do I need to run setup.py to compile and create the extension for pathFinder.cpp? If you could detail some of the intermediate steps to start training(eg. building the pathFinder module), that would be great!

**Comment by xphoniex** [comment]
> Thanks for the reply. I assume I need to re-install torch with python2.7 also?
> 
> Should it be able to start training off the bat by running python main.py? Or do I need to run setup.py to compile and create the extension for pathFinder.cpp? If you could detail some of the intermediate steps to start training(eg. building the pathFinder module), that would be great!

Yup, you need to install torch for python2.7 as well.

pathFinder is already compiled for you [here](https://github.com/xphoniex/alphazero-quoridor/blob/master/quoridor/pathFinder.so) and you should be able to run the code out of the box on a Linux machine.

Haven't tried any other OS but have you seen [setup.py](https://github.com/xphoniex/alphazero-quoridor/blob/master/quoridor/pathFinder-module/setup.py):
```
# $ python setup.py build
# $ sudo python setup.py install
# $ sudo rm -rf build
...
```

**Comment by 2020chris** [comment]
Yes it started training. Is it possible to run on GPU currently? Since I was having compatibility issues with Python2.7, PyTorch 1.4, and CUDA 11.4 in my conda env.

Also, can you explain how you structured/represented the board you pass to the QuoridorNNet, specifically the dimensions 4xnxn of the pieces field of the Board class? Thanks.

**Comment by xphoniex** [comment]
> Is it possible to run on GPU currently?

Yes, you can train on GPU, I trained my own model using Colab.

> Also, can you explain how you structured/represented the board you pass to the QuoridorNNet, specifically the dimensions 4xnxn of the pieces field of the Board class? 

You're gonna have to more specific.

**Comment by Mihonarium** [comment]
If you're still interested, I updated the code so it works with python3: https://github.com/Mihonarium/alphazero-quoridor
